bindings/nim/kzg.nim

```diff
@@ -61,71 +68,88 @@ template verify(res: KZG_RET, ret: untyped): untyped =
 # Public functions
 ##############################################################
 
-proc loadTrustedSetup*(input: File): Result[KzgCtx, string] =
+proc loadTrustedSetup*(input: File, precompute: Natural): Result[KzgCtx, string] =
   let
     ctx = newKzgCtx()
-    res = load_trusted_setup_file(ctx.val, input)
+    res = load_trusted_setup_file(ctx.val, input, precompute.csize_t)
   verify(res, ctx)
 
-proc loadTrustedSetup*(fileName: string): Result[KzgCtx, string] =
+proc loadTrustedSetup*(fileName: string, precompute: Natural): Result[KzgCtx, string] =
   try:
     let file = open(fileName)
-    result = file.loadTrustedSetup()
+    result = file.loadTrustedSetup(precompute)
     file.close()
   except IOError as ex:
     return err(ex.msg)
 
-proc loadTrustedSetup*(g1: openArray[G1Data],
-                       g2: openArray[G2Data]):
+proc loadTrustedSetup*(g1MonomialBytes: openArray[byte],
+                       g1LagrangeBytes: openArray[byte],
+                       g2MonomialBytes: openArray[byte],
+                       precompute: Natural):
                          Result[KzgCtx, string] =
-  if g1.len == 0 or g2.len == 0:
+  if g1MonomialBytes.len == 0 or g1LagrangeBytes.len == 0 or g2MonomialBytes.len == 0:
     return err($KZG_BADARGS)
 
   let
     ctx = newKzgCtx()
     res = load_trusted_setup(ctx.val,
-      g1[0][0].getPtr,
-      g1.len.csize_t,
-      g2[0][0].getPtr,
-      g2.len.csize_t)
+      g1MonomialBytes[0].getPtr,
+      g1MonomialBytes.len.csize_t,
+      g1LagrangeBytes[0].getPtr,
+      g1LagrangeBytes.len.csize_t,
+      g2MonomialBytes[0].getPtr,
+      g2MonomialBytes.len.csize_t,
+      precompute.csize_t)
   verify(res, ctx)
 
-proc loadTrustedSetupFromString*(input: string): Result[KzgCtx, string] =
+proc loadTrustedSetupFromString*(input: string, precompute: Natural): Result[KzgCtx, string] =
   const
+    NumG1 = FIELD_ELEMENTS_PER_BLOB
     NumG2 = 65
-    G1Len = G1Data.len
-    G2Len = G2Data.len
+    G1Len = 48
+    G2Len = 96
 
   var
     s = newStringStream(input)
-    g1: array[FIELD_ELEMENTS_PER_BLOB, G1Data]
-    g2: array[NumG2, G2Data]
+    g1MonomialBytes: seq[byte] = newSeq[byte](NumG1 * G1Len)
+    g1LagrangeBytes: seq[byte] = newSeq[byte](NumG1 * G1Len)
+    g2MonomialBytes: seq[byte] = newSeq[byte](NumG2 * G2Len)
```

Comment:
I think `NumG2 * G2Len` is small enough to put in stack. `newSeq` is slow.

Reply:
Makes sense. For consistency, I did it for all. It's ~0.5 MiB in total. Should be fine.